### ISSUE TITLE
Subscene provider stopped working

### ISSUE THREAD

**Issue by A-Alhashem** [morpheus65535/bazarr#459]
**Describe the bug**
Subscene provider was working with the development branch v0.7.5. However, 2days ago it stopped working as it is no longer fetching subtitles, and will not show up with manual search.

**To Reproduce**
1. When having subscene as one of the providers, choose any movie/TV show
2. Click on manual search
3. All included providers will show except Subscene

**Expected behavior**
Subscene should've been included with automatic and manual search. I have also used Anti-Captcha with Subscene, it was working previously.

**Software (please complete the following information):**
 - Bazarr: v 0.7.5
 - Radarr v 0.2.0.1344
 - Sonarr v 2.0.0.5322
 - OS: Windows 10 64bit

**Comment by morpheus65535** [comment]
@Pannal you want to take a look please?

**Comment by morpheus65535** [comment]
Can you test dev branch which include a fix for this?

**Comment by A-Alhashem** [comment]
Will test it. Anti-Captcha changed their ways, as of right now they need proof of document/id to be sent to them for their services to work.
but that's a problem for another day, just need to change to death by captcha provider to work with subscene.

will reply back if it works

**Comment by A-Alhashem** [comment]
Just to update, I've tried again by using Death By Captcha, and searched again for subscene provider. I didn't receive any subtitle from subscene with manual search.
Here's a copy of subscene log.

10/06/2019 22:42:54|DEBUG   |urllib3.connectionpool          |https://subscene.com:443 "POST /subtitles/searchbytitle HTTP/1.1" 400 None|

10/06/2019 22:42:54|ERROR   |subliminal_patch.core           |Unexpected error in provider 'subscene': Traceback (most recent call last):  File "c:\bazarr\bazarr\../libs/subliminal_patch\core.py", line 162, in list_subtitles_provider    results = self[provider].list_subtitles(video, provider_languages)  File "c:\bazarr\bazarr\../libs/subliminal_patch\providers\subscene.py", line 151, in list_subtitles    return [s for s in self.query(video) if s.language in languages]  File "c:\bazarr\bazarr\../libs/subliminal_patch\providers\subscene.py", line 213, in query    film = search(term, session=self.session, release=False, throttle=self.search_throttle)  File "c:\bazarr\bazarr\../libs/subscene_api\subscene.py", line 280, in search    session=session)  File "c:\bazarr\bazarr\../libs/subscene_api\subscene.py", line 74, in soup_for    ret.raise_for_status()  File "c:\bazarr\bazarr\../libs/requests\models.py", line 940, in raise_for_status    raise HTTPError(http_error_msg, response=self)HTTPError: 400 Client Error: Bad Request for url: https://subscene.com/subtitles/searchbytitle|Traceback (most recent call last):  File "c:\bazarr\bazarr\../libs/subliminal_patch\core.py", line 162, in list_subtitles_provider    results = self[provider].list_subtitles(video, provider_languages)  File "c:\bazarr\bazarr\../libs/subliminal_patch\providers\subscene.py", line 151, in list_subtitles    return [s for s in self.query(video) if s.language in languages]  File "c:\bazarr\bazarr\../libs/subliminal_patch\providers\subscene.py", line 213, in query    film = search(term, session=self.session, release=False, throttle=self.search_throttle)  File "c:\bazarr\bazarr\../libs/subscene_api\subscene.py", line 280, in search    session=session)  File "c:\bazarr\bazarr\../libs/subscene_api\subscene.py", line 74, in soup_for    ret.raise_for_status()  File "c:\bazarr\bazarr\../libs/requests\models.py", line 940, in raise_for_status    raise HTTPError(http_error_msg, response=self)HTTPError: 400 Client Error: Bad Request for url: https://subscene.com/subtitles/searchbytitle|

**Comment by morpheus65535** [comment]
@pannal I've just tried on my own instance of Bazarr and I can reproduce it:

```
Unexpected error in provider 'subscene': 
Traceback (most recent call last):
  File "/app/bazarr/bazarr/../libs/subliminal_patch/core.py", line 162, in list_subtitles_provider
    results = self[provider].list_subtitles(video, provider_languages)
  File "/app/bazarr/bazarr/../libs/subliminal_patch/providers/subscene.py", line 151, in list_subtitles
    return [s for s in self.query(video) if s.language in languages]
  File "/app/bazarr/bazarr/../libs/subliminal_patch/providers/subscene.py", line 213, in query
    film = search(term, session=self.session, release=False, throttle=self.search_throttle)
  File "/app/bazarr/bazarr/../libs/subscene_api/subscene.py", line 280, in search
    session=session)
  File "/app/bazarr/bazarr/../libs/subscene_api/subscene.py", line 74, in soup_for
    ret.raise_for_status()
  File "/app/bazarr/bazarr/../libs/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)HTTPError: 400 Client Error: Bad Request for url: https://subscene.com/subtitles/searchbytitle
```

Of course this URL doesn't seems to be valid. Can you take a look at it? Thanks.

**Comment by pannal** [comment]
How is this url not valid? That's their new POST endpoint. Unfortunately they require recaptcha everywhere now.

I'll take a look once I get back.

**Comment by morpheus65535** [comment]
That’s what I get when I try to load the page:
![Image](https://i.imgur.com/lpuUM9o.jpg)

**Comment by pannal** [comment]
Interesting. They might change their endpoints regularly now. I'll take a look soon

**Comment by morpheus65535** [comment]
Thanks!

**Comment by A-Alhashem** [comment]
Thank you

**Comment by morpheus65535** [comment]
@pannal Did you had time to take a look at this one?

**Comment by pannal** [comment]
A little. This is getting serious.

**Comment by morpheus65535** [comment]
Can you try the dev branch? Don't forget to define a username and password for Subscene in settings and provide an anti-captcha provider.

**Comment by pannal** [comment]
The dev branch doesn't have the latest fixes yet. I'm still testing it locally.

**Comment by A-Alhashem** [comment]
lol. Thanks for the efforts =) will give a an update to the latest dev branch and test it out.

**Comment by A-Alhashem** [comment]
Subscene now works with the new update! Although its taking around 40-50sec for it to load the subtitles (when manually searching).
when checking the logs, it is not using captchas as of testing it at this moment (also checked with my captcha provider), and in logs, subscene says its "Re-using old subscene cookies:" dunno if its relevant or not.

Thanks again!

**Comment by A-Alhashem** [comment]
[bazarr (1).log](https://github.com/morpheus65535/bazarr/files/3315526/bazarr.1.log)

this is the debug log.

**Comment by morpheus65535** [comment]
@pannal Check his log please. Why Subscene provider return subtitles for a lot more language that what was requested? It's clearly not optimal. We are hitting their website a lot and parsing way to much HTML...

**Comment by pannal** [comment]
I don't have the faintest clue why he searches without a language filter tbh.

**Comment by A-Alhashem** [comment]
@pannal Just came back from work, and tried a different approach, it may sound stupid but I actually went into the site itself subscene.com and used their filter, to only filter 2 languages I needed. That's the only thing different from yesterday's test and today. And it worked.

[bazarr (5).log](https://github.com/morpheus65535/bazarr/files/3317757/bazarr.5.log)

from the logs, it re-used my old cookies, and fetched the filtered languages. I know bazarr uses API to fetch, and it may have no relation to the actual website's filters.. I have no explanation lol

I used the same exact media file I used before, the same file was tested now and yesterday.
The search took approximately 20-30secs, more than half the time than before (as there was a lot of different subtitles on the same language filtered for the same media, which is expected I guess)

I'll try more tests on different media's, And will see the logs. Maybe the last test had a cached bug or something dunno.

**Comment by A-Alhashem** [comment]
Ok this time I went back to subscene.com and unfiltered from the site itself, tried the test again and now the log is fetching every language there is from the same media file I used.

here's the log:

[bazarr (6).log](https://github.com/morpheus65535/bazarr/files/3317763/bazarr.6.log)

it took around 45secs (give or take) for it to fetch manual search via bazarr.

So i'm assuming with subscene, a temp fix would be that the user has to go on the site and filter the language they need.

Hope this helps out
